code-examples/protect-page-login/nextjs/app/auth/registration/page.tsx

```diff
@@ -0,0 +1,27 @@
+// Copyright © 2024 Ory Corp
+// SPDX-License-Identifier: Apache-2.0
+
+import { Registration } from "@ory/elements-react/theme"
+import { enhanceOryConfig } from "@ory/nextjs"
+import { getRegistrationFlow, OryPageParams } from "@ory/nextjs/app"
+
+import baseConfig from "@/ory.config"
+
+export default async function RegistrationPage(props: OryPageParams) {
```

Comment:
With these files we are also allowing the customers to bring their own ui - so implement their own registration page. In my view, this should go into another guide like "Build your own UI" (or something similar). See this section: https://www.ory.sh/docs/getting-started/custom-ui

Reply:
The nextjs quickstart should focus on "what do i need to do to authenticate nextjs with ory"?

For custom styling etc we should put it into another section

Reply:
After giving this some thought, I think we should leave it this way. It aligns with the Next.js way of doing auth (using the `/auth` route). I think the other option is to use the FrontendAPI, which while simpler, does not provide a Next.js agnostic way of doing auth.

Here are some examples:
[Auth.js](https://authjs.dev/getting-started/installation?framework=Next.js)
[better-auth](https://www.better-auth.com/docs/integrations/next#create-api-route)